internal/docs/markdown.go

```diff
@@ -46,17 +46,43 @@ func hasNonHelpFlags(fs *pflag.FlagSet) (found bool) {
 	return
 }
 
+var hiddenFlagDefaults = map[string]bool{
+	"false": true,
+	"":      true,
+	"[]":    true,
+	"0s":    true,
+}
+
+var defaultValFormats = map[string]string{
+	"string":   " (default \"%s\")",
+	"duration": " (default \"%s\")",
+}
+
+func getDefaultValueDisplayString(f *pflag.Flag) string {
+
+	if hiddenFlagDefaults[f.DefValue] || hiddenFlagDefaults[f.Value.Type()] {
+		return ""
+	}
+
+	if dvf, found := defaultValFormats[f.Value.Type()]; found {
+		return fmt.Sprintf(dvf, f.Value)
+	}
+	return fmt.Sprintf(" (default %s)", f.Value)
```

Comment:
This can be simplified, removing the need for package variables:

```suggestion
	switch f.Value.Type() {
	case "string":
		return fmt.Sprintf(" (default \"%s\")", f.Value)
	case "duration":
		return fmt.Sprintf(" (default \"%s\")", f.Value)
	default:
		return fmt.Sprintf(" (default %s)", f.Value)
	}
```

Reply:
Should ```hiddenFlagDefaults``` be removed also then? I think removing them makes this a bit messy but after reading up on why people don't like package level variables I get it.

Reply:
I wasn't thinking about it at the time but I suppose you could, yeah 🤔